charters/charter-template.md

```diff
@@ -0,0 +1,30 @@
+
+# Project Name
+
+
+## Description
+
+## Objective
+
+## In-scope
+
+## Out-of-scope
+
+## Milestones
```

Comment:
Can you talk a little more about milestones and why they would be in a charter? I can see a general definition of functionality that describes the new process but the milestones would change and grow and seem to be something that would be addressed iteratively in other documents.

Reply:
Milestones for software projects are recommended for oversight. A charter is a contract with the HCA DCP community, which commits the project owner to meet explicit high-level milestones and deliver specific “products”.  And once the milestones are achieved, the project owner can either renegotiate the charter or conclude. If there is a failure to deliver, then the charter and ownership can be reviewed. 

For governance charters such as PM++ and Architecture++, there would be no milestones.

Reply:
Most standards efforts last 2-3 years which I consider long-term. I would recommend chartering for shorter periods and re-chartering based on updated scope. OASIS requires a re-charter if the scope of a technical committee changes.  You might update the Secondary Analysis charter when new pipelines are approved by HCA Science governance. 

If you strongly disagree, I would recommend reviewing with the other Leads to see if there is a consensus position among you.

Reply:
I agree with Tim that listing milestones here is anti-agile. I could see writing the vision of the product and a link to a product roadmap.

Reply:
- Re. agile: @kbergin it would be anti-agile if charters remain static. Comparatively, @brianraymor suggested above re-chartering if scope is updated. We can address your core concern--that modules can adapt their scope fluidly when requirements change--if we make sure that the approval process for amending charters is responsive.
- Re. oversight and transparency: making delivery accountabilities clear is a good idea. As mentioned, it's ok to change milestones, but dependent teams should be informed in a transparent way. Making milestones and changes to them visible here accomplishes that.

Reply:
If we view milestones as product vision statements and not a list of specific deliverables then I don't see them conflicting with Agile development principles. But it is confusing to call these "milestones" as those are usually used to refer to inflection points along the effort to develop something larger.   So how about calling them "Project Vision" or "Project Goals"?

Reply:
I'm in favour of milestones as key points of alignment between components. I think this is what we have so far being trying to accomplish with our quarterly goals and roadmapping, and I'd like to have a common vision for the whole DCP that can be mapped into milestones in modules. I think allowing these to be broader in scope with less specific detail provides opportunities for evolution that *encourages* agility rather than stifles it, especially coupled with a continuous review process. I'm ambivalent about whether we call them milestones or goals.